pycbc/filter/qtransform.py

```diff
@@ -32,12 +32,156 @@
 
 """
 
+
 from numpy import pi, ceil, log, exp
 import numpy as np
 from pycbc.strain  import next_power_of_2
 from pycbc.types.timeseries import FrequencySeries, TimeSeries
+from scipy.interpolate import (interp2d, InterpolatedUnivariateSpline)
+from pycbc.fft import ifft
+from pycbc.types import zeros
 from numpy import fft as npfft
 
+from matplotlib import use
+use('Agg')
+from matplotlib import pyplot as plt
+from matplotlib.colors import BoundaryNorm
+from matplotlib.ticker import MaxNLocator
+from matplotlib.pyplot import specgram
+
+def plotter(interp, out_dir, now, frange, fseries, sampling, tres=0.001, fres=1.):
+    """Plotting mechanism for pycbc spectrograms
+
+    Parameters
+    ----------
+    interp:
+        2D interpolation function
+    out_dir:
+        path to output directory
+    now:
+        unique label for output directory
+    frange:
+        upper and lower bounds on frequency range plotted
+    fseries: 'pycbc FrequencySeries'
+        frequency series of data set
+    sampling:
+        sampling rate of data set
+    tres:
+        desired time resolution
+    fres:
+        desired frequency resolution
+
+    Returns
+    -------
+    plt:
+        matplotlib spectrogram figure
+
+    """
+
+    # plot a spectrogram of the q-plane with the loudest normalized tile energy
+    print 'plotting ...'
+    # intitialize variables
+    xarr=np.linspace(0, 1, 1. / tres)
+    yarr=np.arange(int(frange[0]), int(frange[1]), fres)
+    z = interp(xarr,yarr)
+
+    # pick the desired colormap
+    cmap = plt.get_cmap('viridis')
+
+    fig = plt.plot()
+
+    p1 = plt.pcolormesh(xarr,
+                        yarr,
+                        z, cmap=cmap, norm=None)
+    plt.colorbar(p1)
+    plt.title('Pycbc q-transform')
+
+    plt.savefig('%s/run_%s/spec.png' % (out_dir,now))
+
+    return plt
+
+def Qplane(qplane_tile_dict, fseries, sampling, out_dir, now, frange, normalized=True, tres=0.001, fres=1.):
+    """Performs q-transform on each tile for each q-plane and selects 
+       tile with the maximum normalized energy. Q-transform is then 
+       interpolated to a desired frequency and time resolution.
+
+    Parameters
+    ----------
+    qplane_tile_dict:
+        Dictionary containing a list of q-tile tupples for each q-plane
+    fseries: 'pycbc FrequencySeries'
+        frequency-series data set
+    sampling:
```

Comment:
I don't think you need sampling as an argument. That can be derived from the fseries itself. sample_rate = (len(fseries) -1) * 2 * df